File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed) => speed switch
    {
        < 1 => 0.0,
        < 5 => 1.0,
        < 9 => 0.9,
        9 => 0.8,
        > 9 => 0.77,
    };

    public static double ProductionRatePerHour(int speed) => SuccessRate(speed) * speed * 221.0;

    public static int WorkingItemsPerMinute(int speed) => (int) ProductionRatePerHour(speed) / 60;
}

File: solutions/julia/cars-assemble/1/cars-assemble.jl
function success_rate(speed)
    if speed ≤ 0
        return 0
    elseif 1 ≤ speed ≤ 4
        return 1
    elseif  5 ≤ speed ≤ 8
        return 0.9
    elseif speed == 9
        return 0.8
    else
        return 0.77
    end
end

const cars_ratio = 221
function production_rate_per_hour(speed)
    speed * cars_ratio * success_rate(speed) 
end

function working_items_per_minute(speed)
    Int64(production_rate_per_hour(speed) ÷ 60)
end
